### PR TITLE
Add support for relative path for nested include

### PR DESCRIPTION
**Pull request by loicgasser**
Not sure what you tried to accomplish with cwd, append absolute paths?

According to me in mapfile you have either
- relative paths
- absolute paths

It seems that it was overriden every time when calling `parse_file`, which is not something I'd expect and I couldn't find any test for cwd argument.

Hope that you are happy with these changes.

In the utils section I added all the options you can give the parser (namely `add_linebreaks` for `loads` and `expand_includes` + `add_linebreaks` for `load`)

DOC:
http://mapserver.org/mapfile/include.html

```
Supported in versions 4.10 and higher.

The name of the file to be included MUST be quoted (single or double quotes).

Includes may be nested, up to 5 deep.

File location can be given as a full path to the file, or (in MapServer >= 4.10.1) as a path relative to the mapfile.
```

### CODE REVIEW

**Comment by geographika** [comment]
@loicgasser - I'll need to get some time to review this in more detail. Thanks for the comments and pull request. From memory I was using the `cwd` parameter so that I could load a mapfile from a string with relative includes, and load these from disk.

**Comment by loicgasser** [comment]
ok thanks, I can adapt/change it if you guide me.

File: mappyfile/parser.py [hunk 1]
```diff
@@ -66,11 +68,9 @@ def open_file(self, fn):
             raise
 
     def parse_file(self, fn):
-
-        self.cwd = os.path.dirname(fn)
```
Comment (loicgasser) [inline mappyfile/parser.py]: main issue was that. always the same cwd which was overriden here

File: mappyfile/parser.py [hunk 2]
```diff
@@ -9,52 +9,54 @@
...
         for idx, l in enumerate(lines):
-            if l.strip().lower().startswith('include'):
-                if '#' in l:
-                    l = l[:l.index('#')]
-
-                parts = [p for p in l.split()]
-
-                assert (len(parts) == 2)
-                assert (parts[0].lower() == 'include')
-                fn = os.path.join(self.cwd, self.strip_quotes(parts[1]))
```
Comment (loicgasser) [inline mappyfile/parser.py]: and as a result this would not follow relative nested paths.

**Comment by geographika** [comment]
@loicgasser - thanks for this. Merged. I increased the line length for flake8 as the default of 80 is quite low.

**Pull request merged**